### PR TITLE
[stable/kubernetes-dashboard] allow extra containers

### PR DESCRIPTION
**Pull request by sylwit**
#### What this PR does / why we need it:

This PR allow the usage of extraContainers. It's useful if you need to access the dashboard behind a oAuth proxy. @desaintmartin 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

File: stable/kubernetes-dashboard/values.yaml [hunk 1]
```diff
@@ -164,3 +164,20 @@ podDisruptionBudget:
   enabled: false
   minAvailable:
   maxUnavailable:
+
+## Enable an Specify container in extraContainers. This is meant to allow adding an authentication proxy to a dashboard pod
+extraContainers: |
```
Comment (kfox1111) [inline stable/kubernetes-dashboard/values.yaml:172]: Should this just be yaml rather then a string?

File: stable/kubernetes-dashboard/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: kubernetes-dashboard
-version: 1.5.2
+version: 1.5.3
```
Comment (kfox1111) [inline stable/kubernetes-dashboard/Chart.yaml]: Its a new feature, so should probably be 1.6.0.

**Review by kfox1111 — COMMENTED** [review]
Looks pretty good. one minor issue and a question.

**Comment by desaintmartin** [comment]
Out of curiosity, what is the use case behind this ? I use oauth2-proxy with k8s-dash as well. This complexifies the chart so it should really be a feature.

**Comment by sylwit** [comment]
If you use oauth2-proxy you probably create another service for the proxy.
By starting the proxy container in the same pod as the dashboard container
you need only one service.

This is taken from the grafana chart and I found it useful and easy to use
so I thought it was a nice little feature to add.

**Comment by kfox1111** [comment]
I'm ok adding the option to add more containers.

I too have been using the oauth2_proxy chart plugged into nginx-ingress as a auth source to front the dashboard though. so not strictly needed in this case.

**Comment by sylwit** [comment]
There is one thing I'm not fan of is this template

```
{{- if hasKey .Values.service "targetPort" }}
    targetPort: {{ .Values.service.targetPort }}
    name: "TCP"
{{- else if .Values.enableInsecureLogin }}
    targetPort: 9090
    name: "http"
{{- else }}
    targetPort: 8443
    name: "https"
{{- end }}
```
if you use a targetPort, enableInsecureLogin will have no effect and it's not enough explicit I think. Any ideas or thoughts about that ?

**Comment by desaintmartin** [comment]
This can be documented, indeed! Thanks for the catch.

**Comment by sylwit** [comment]
I added a comment to be more explicit. If it's ok for you when can you merge it ?
Thank you

**Comment by sylwit** [comment]
@desaintmartin @kfox1111 any comment on this PR ? If not, can you merge it please ?

**Comment by desaintmartin** [comment]
I think there a still a comment that is not answered.
Moreover, would you mind adding a small test file in ci directory about that? see what is done in elasticsearch for example.
/ok-to-test

**Pull request closed without merging**